### PR TITLE
Enable strict mode and surface errors to the player

### PR DESCRIPTION
**Pull request by lazeratops**
This PR enables all TypeScript strictness checks, and updates the code to fix the new compiler errors that come with these.

It also adds some basic error surfacing to the client in the form of a popup like this:

image

### CODE REVIEW

File: src/client/game/board.ts [hunk 1]
```diff
@@ -261,6 +274,10 @@ export class Board {
       subheading = "";
     }
     const gameStatus = document.getElementById("gameStatus");
+    if (!gameStatus) {
+      showGameError(new ErrGeneric("Game status DOM element not found"));
```
Comment (lazeratops) [inline src/client/game/board.ts:278]: showGameError() is a new method to surface errors to the client within game UI. I will highlight its implementation as a separate comment below.

File: src/client/game/errors/errGeneric.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { GameError } from "./error";
+
+export default class ErrGeneric extends Error implements GameError {
```
Comment (lazeratops) [inline src/client/game/errors/errGeneric.ts:3]: Some new errors are defined in this PR; most of these are unrecoverable (meaning the user will be prompted to try and re-join the game/refresh the page). `ErrGeneric` is the most generic with these, allowing the caller to specify whether it is recoverable or not (no by default).

File: .eslintrc.json [hunk 3]
```diff
@@ -14,7 +14,11 @@
   "rules": {
     "prettier/prettier": ["error"],
     "react/jsx-filename-extension": "off",
-    "@typescript-eslint/no-use-before-define": "off"
+    "@typescript-eslint/no-use-before-define": "off",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      { "argsIgnorePattern": "^_" }
```
Comment (ablwr) [inline .eslintrc.json:20]: ^________________^

**Review by ablwr — APPROVED** [review]
Ummmm I think this looks great!

**Review by nienkedekker — APPROVED** [review]
🚀

**Review by karlstolley — APPROVED** [review]
Some really nice-looking work here.

**Pull request merged**